Add conanfile

Ujifman opened the pull request:
Added conanfile.py for building as Conan package, with linked dependencies
It is good idea to push xlsxio to center.conan.io in future as Conan package

brechtsanders commented:
`FindEXPAT.cmake` is part of CMake, why do you rewrite it in your pull request?

Also, what is the reason the line:
```
INCLUDE_DIRECTORIES(${EXPAT_INCLUDE_DIRS})
```
was changed to:
```
INCLUDE_DIRECTORIES(${EXPAT_INCLUDE_DIR})
```

As for Conan, I don't have any experience with that (yet), but I'm okay with that if you have tested your files on different platforms supported by Conan.

Ujifman commented:
@brechtsanders 
1. I took `FindEXPAT.cmake` from CMake distrib and modified it, because by default it cannot find expat lib with `d` suffix (debug build on Windows), so I added this opportunity
2. I changed `EXPAT_INCLUDE_DIRS` on `EXPAT_INCLUDE_DIR`, because if you build without Conan, then CMake build fails, because FindEXPAT creates `EXPAT_INCLUDE_DIR` variable, and your CMakeLists looks for `EXPAT_INCLUDE_DIRS` variable

brechtsanders commented:
Wouldn't it make more sense to report to the CMake developer(s) that `FindEXPAT.cmake` needs some tweaking for debug builds on Windows?

`FindEXPAT.cmake` contains:
```cmake
if(EXPAT_FOUND)
  set(EXPAT_LIBRARIES ${EXPAT_LIBRARY})
  set(EXPAT_INCLUDE_DIRS ${EXPAT_INCLUDE_DIR})
```
so if Expat is found `EXPAT_INCLUDE_DIRS` is definitely defined. Is it possible there was another issue?

Ujifman commented:
I'l try to explain chronologically.
At first is tried to build xlsxio manually (v0.2.31). 
1. I found dependencies in Conan Center, installed and built them. `expat`, `minizip`, `zlib`
1. Then I executed CMake configure, and filled out `EXPAT_DIR`, `MINIZIP_DIR`, `ZLIB_DIR` variable. I've used CMake GUI on Windows
1. Then I executed CMake configure again. `minizip` and `zlib` were successfully found, but `expat` still showed error
    ![image](https://user-images.githubusercontent.com/34909951/165990409-6b033982-f665-4484-8628-bc1c9bb3f123.png)
1. I started research to find out why it is still not found, and concluded, that CMake expat module not works, because it is not standart location, but in `xlsxio` CMakeLists.txt, one uses `find_path` and write value to variable `EXPAT_INCLUDE_DIR` but then includes variable `EXPAT_INCLUDE_DIRS`, and that is the problem, why CMake cannot find `expat`
    ![image](https://user-images.githubusercontent.com/34909951/165990803-cf0b8057-b072-4070-80ec-980a7b134ead.png)
1. Then I fixed variable name in include to `EXPAT_INCLUDE_DIR` and successfully built `xlsxio`

On this point i created first [pull request](https://github.com/brechtsanders/xlsxio/pull/108)

---

After that I created Conan package, for my purposes and decided that it will be usefull for somebody.
When I started to build it through Conan, I found out that Conan uses default `find_package` scripts of CMake. And on Windows expat builded in Debug mode has name `libexpatd.dll`, with `d` suffix, and default CMake `find_package` cannot recognize library. Thats why I added modified `FindEXPAT.cmake` here, for Conan builds to be able to find expat.

brechtsanders commented:
#108 already fixes `CMakeLists.txt` to use `EXPAT_INCLUDE_DIR` instead of `EXPAT_INCLUDE_DIRS`

uilianries commented:
Hello!

I just want to inform Xlsxio is available on the official Conan Center already: https://conan.io/center/recipes/xlsxio?version=0.2.34

In case you want to add it to your documentation, I can open a separate PR as well.

Still, you can add a conanfile in this project to solve its dependencies like libzip and minizip-ng. 

Regards!
